chain/consensus/hierarchical/actors/subnet/subnet_actor.go

```diff
@@ -203,8 +204,13 @@ func (a SubnetActor) Leave(rt runtime.Runtime, _ *abi.EmptyValue) *abi.EmptyValu
 	priorBalance := rt.CurrentBalance()
 	var retFunds abi.TokenAmount
 	rt.StateTransaction(&st, func() {
-		// Remove stake from stake balanace table.
+		// Remove stake from stake balance table.
 		retFunds = st.rmStake(rt, sourceAddr, stakes, minerStake)
+
+		// Remove the validator to signal to other validators on the subnet to run reconfiguration.
+		if st.ValidatorSet != nil {
```

Comment:
We should remove `MinerList` in favor of `ValidatorSet` and include all the information in `ValidatorSet` (so you don't have to do this check). In any case, not worth changing it here because soon we will deprecate these actors, and we'll only have the FVM actors, so we need to implement this functionality in the Rust actor. This approach LGTM.

Reply:
ok